frontend/packages/console-app/src/components/network-policies/network-policy-model.ts

```diff
@@ -66,81 +67,82 @@ const factorOutError = <T>(list: (T | ConversionError)[]): T[] | ConversionError
   return list as T[];
 };
 
-const selectorToK8s = <T>(
-  selector: string[][],
-  emptyValue: Selector | undefined,
-): Selector | undefined | ConversionError => {
+const errors = {
```

Comment:
I am always hesitant to put such strict error checking in the client since we've been bitten many times with APIs changing. Even maintaining backwards compatibility, additional properties can be added or validation loosened on the backend. We're also reimplementing a lot of backend logic.

Having said that, I don't know that there is a way around it. I think it's necessary if we want users to toggle back and forth to avoid losing data.

Reply:
I think I understand, however here these checks are performed not while sending data to the backend, but on reading  data before loading it into the form. Useful for instance to throw an error when coming from the YAML editor and being unable to load it into the form.